### PR TITLE
added more common types matchers

### PR DESCRIPTION
**Pull request by gonzalobustosglob**
# Description

This PR provides matchers for PublicKey, Int64, UInt64 and UInt32 classes and their tests.

### CODE REVIEW

**Comment by AlanVerbner** [comment]
@gonzalobustosglob pls add a PR description 🙏

File: lib/matchers/common-types.ts [hunk 1]
```diff
@@ -67,5 +67,80 @@ export default () => {
...
       };
     },
+    /**
+     * Checks if the actual value is equal to the expected value for a given field.
+     * @template T - The type of the field.
+     * @param {T & State<T>} actual - The actual field value to compare.
+     * @param {PrivateKey} value - The expected field value to compare.
+     * @returns {MatcherResult} - The result of the comparison.
+     */
+    toEqualPrivateKey<T extends PrivateKey>(actual: T & State<T>, value: PrivateKey) {
```
Comment (AlanVerbner) [inline lib/matchers/common-types.ts]: Does it make sense to store in the contract state the private key? 🤔

File: lib/matchers/common-types.ts [hunk 2]
```diff
@@ -67,5 +67,80 @@ export default () => {
...
+     * @template T - The type of the field.
+     * @param {T & State<T>} actual - The actual field value to compare.
+     * @param {Int64Like} value - The expected field value to compare.
+     * @returns {MatcherResult} - The result of the comparison.
+     */
+    toEqualInt64<T extends Int64>(actual: T & State<T>, value: Int64Like) {
+      /* Check if actual is a State object */
+      const toMatch: Int64 = extractFromState(actual);
+
+      if (!(toMatch instanceof Int64)) {
```
Comment (AlanVerbner) [inline lib/matchers/common-types.ts]: what about `UInt32`, `UInt64`?

File: lib/matchers/common-types.ts (hunk 1 above)
Comment (gonzalobustosglob) [reply lib/matchers/common-types.ts]: It doesn't. Removed.

File: lib/matchers/common-types.ts (hunk 2 above)
Comment (gonzalobustosglob) [reply lib/matchers/common-types.ts]: Added them.

**Pull request merged**